distribution/openrct2.d.ts

```diff
@@ -2606,6 +2606,51 @@ declare global {
         "waiting_to_depart" |
         "waiting_to_start";
 
+    type CrashParticleType =  0 | 1 | 2 | 3 | 4;
+
+    /**
+     * Represents a vehicle explosion particle. They are emitted during a vehicle
+     * crash and will bounce until their timer expires and they are automatically
+     * removed.
+     */
+    interface CrashedVehicleParticle extends Entity {
+        /**
+         * The colour of the particle. Only body and trim are used.
+         */
+        colours: VehicleColour;
+
+        /**
+         * The lifetime of the particle in ticks. Default value 65535. Entity is
+         * automatically removed at 0.
+         */
+        timeToLive: number;
+
+        /**
+         * The particle velocity.
+         */
+        velocity: CoordsXYZ;
+
+        /**
+         * The x-component of the particle acceleration.
+         */
+        acceleration: CoordsXYZ;
+
+        /**
+         * The type of crash particle.
+         */
+        spriteBase: CrashParticleType;
+
+        /**
+         * The current frame of the crash particle.
+         */
+        frame: number;
+
+        /**
+         * Sets the initial values of the crash particle using the game's default values.
+         */
+        setupValues(colours: VehicleColour): void;
```

Comment:
I'm not fully understanding this one's purpose? Is this for setting the global default particle colours?

Reply:
This function calls the c++ function `CrashedVehicleParticle::SetupValues` which applies a random sprite type, a random frame number, and random acceleration. It's the same function the game calls when a vehicle crashes.

Reply:
Maybe it's better to name it `launch()` `launchRandomly()` on the plugin side then to make its purpose more clear?